src/Adapter/CartRule/Repository/CartRuleRepository.php

```diff
@@ -328,6 +329,76 @@ public function restrictCartRules(CartRuleId $cartRuleId, array $restrictedCartR
         );
     }
 
+    /**
+     * @param CartRuleId $cartRuleId
+     * @param CountryId[] $restrictedCountryIds
+     *
+     * @return void
+     */
+    public function setCountryRestrictions(CartRuleId $cartRuleId, array $restrictedCountryIds): void
+    {
+        $this->removeCountryRestrictions($cartRuleId);
+
+        if (empty($restrictedCountryIds)) {
+            return;
+        }
+
+        $checkedIds = [];
+        $insertValues = [];
+        foreach ($restrictedCountryIds as $restrictedCountryId) {
+            // skip duplicate ids if for some reason they exist
+            if (in_array($restrictedCountryId, $checkedIds, true)) {
```

Comment:
If you want to check duplicate values you should store the int values, if you search by object it will check the instance reference not its value So you could end up with different VOs but same int values

Reply:
oops, I forgot to use getValue() :+1: